File: frontend/app/services/photo-comment.service.ts
/**
 * Created on 15.09.2016.
 */

import {Injectable} from "@angular/core";
import {ApplicationProperties} from "../config/config";
import {Http} from "@angular/http";
import {PhotoComment} from "../models/photo-comment";

@Injectable()
export class PhotoCommentService {
    private properties = new ApplicationProperties();

    constructor(private http: Http) {
    }

    add(comment: PhotoComment) {
        return this.http.post(this.properties.addCommentUrl, comment, {headers: this.properties.authHeader});
    }

    getByPhotoId(id: number) {
        return this.http.get(this.properties.getCommentsByPhotoIdUrl.concat('/' + id), {headers: this.properties.jsonHeader});
    }

}

File: src/main/java/com/example/service/PhotoCommentService.java
package com.example.service;

import com.example.model.Photo;
import com.example.model.PhotoComment;

import java.util.List;

/**
 * Created on 15.09.2016.
 */
public interface PhotoCommentService {
    PhotoComment add(PhotoComment photoComment);

    List<PhotoComment> getByPhoto(Photo photo);
}
